src/main/java/org/icgc/argo/program_service/controller/ProgramController.java

```diff
@@ -0,0 +1,119 @@
+package org.icgc.argo.program_service.controller;
+
+import com.google.protobuf.StringValue;
+import java.io.IOException;
+import java.util.List;
+import java.util.NoSuchElementException;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.icgc.argo.program_service.converter.Grpc2JsonConverter;
+import org.icgc.argo.program_service.model.dto.*;
+import org.icgc.argo.program_service.model.exceptions.NotFoundException;
+import org.icgc.argo.program_service.proto.*;
+import org.icgc.argo.program_service.services.ProgramServiceFacade;
+import org.icgc.argo.program_service.services.auth.AuthorizationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.dao.InvalidDataAccessApiUsageException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@Slf4j
+@RestController
+@RequestMapping("/program")
+public class ProgramController {
+
+  @Autowired private ProgramServiceFacade serviceFacade;
+  @Autowired private Grpc2JsonConverter grpc2JsonConverter;
+  @Autowired private AuthorizationService authorizationService;
+
+  @PostMapping(value = "/createProgram")

```

Comment:
Most endpoints for an entity don't need a custom path, the HTTP request verb should indicate the action.

Create should be a POST request to `/programs`

Reply:
Removed /createProgram from the request mapping